### ISSUE TITLE
Sort out *HelpContent.js files

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#478]
@pixelzoom pointed out to me that it is confusing that some `*HelpContent.js` types extend `HelpContent`, and some extend `HBox`.  Currently the `HelpContent` filename suffix is overloaded. It can mean "a subtype of `HelpContent`, which is often just one of many sections of a keyboard help dialog, and we have also used it for the HBox that contains all of the `HelpContent` instances in the keyboard help dialog.

I propose that we come up with a new suffix for the HBoxes, how about `*HelpContentContainer`.

Two obvious renames would then be:
`FrictionHelpContentPanel` -> `FrictionHelpContentContainer`
`JohnTravoltageKeyboardHelpContent` -> `JohnTravoltageKeyboardHelpContentContainer` 

We should also decide if we should always include `Keyboard` in the name, or if `{{Sim}}HelpContentContainer` is specific enough as a suffix. 

Tagging @jessegreenberg for comment.

**Comment by pixelzoom** [comment]
Various thoughts...

Better naming would help. `HelpContent` should probably be called `KeyboardHelpContent`. Ditto for subclasses like `GeneralNavigationHelpContent`. PhET may have other types of "help" in the future, and you have classes like `JohnTravoltageKeyboardHelpContent`. 

Better documentation would help.  For example, `HelpContent` is described as "Contains help content for a KeyboardHelpDialog". But it's actually not the `helpContent` arg to `KeyboardHelpDialog`, it's a building block to create that content.   

Where `HBox` it used as the superclass, the `align: 'top'` is duplicated and `spacing` is different.  E.g. `JohnTravoltageKeyboardHelpContent` has `align: 'top', spacing: 35`, while `MoleculesAndLightKeyboardHelpContent` has `align: 'top', spacing: 30`.   Seems like a subclass is in order that sets appropriate defaults.

**Comment by pixelzoom** [comment]
Friction has yet-another variation with `FrictionKeyboardHelpContentPanel`, which extends `Panel`.

**Comment by jessegreenberg** [comment]
Another renaming proposal:
 - HelpContent is renamed to KeyboardHelpSection. This Node contains one section of keyboard help content to be used alone or with other content in a KeyboardHelpDialog.
 - {{sim}}KeyboardHelpContent files remain as they are named. These assemble one or more KeyboardHelpSections, and are the content for a KeyboardHelpDialog.

> Seems like a subclass is in order that sets appropriate defaults.

Sounds good, Ill take a look at that too.

**Comment by jessegreenberg** [comment]
@zepumph what do you think about the above? I agree it is beneficial to keep "Keyboard" in the name. If we want to reduce verbosity I would be OK with removing "{{Sim}}" prefix from the KeyboardHelpContent files.

**Comment by zepumph** [comment]
> I would be OK with removing "{{Sim}}" prefix from the KeyboardHelpContent files.

That is unconventional to what we do with all other files like main, config, constants, etc.

I think we should go in the direction of the above proposal. I like almost everything except that I think that all we should prefix all `KeyboardHelpSection` subtypes with the full prefix, i.e. `GeneralNavigationKeyboardHelpSection`. Or shorten just to `GeneralKeyboardHelpSection`.

**Comment by jessegreenberg** [comment]
OK sounds good, based on https://github.com/phetsims/scenery-phet/issues/478#issuecomment-477288070 I will
- [ ] Rename HelpContent to KeyboardHelpSection.
- [ ] Rename GeneralNavigationHelpContent to GeneralKeyboardHelpSection
- [ ] Rename SliderControlsHelpContent to SliderKeyboardHelpSection
- [ ] Rename FrictionKeyboardHelpContentPanel to FrictionKeyboardHelpContent, and change so that it no longer extends Panel.

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/scenery-phet/issues/478#issuecomment-479077697 is done, but some more things need to be renamed like `HelpContentRow` -> `HelpSectionRow` and so on.

**Comment by jessegreenberg** [comment]
Alright, all done in the above commits. Extra commits just renamed local variables in all *HelpContent.js files to match this decision.

Ill take a look at
> Seems like a subclass is in order that sets appropriate defaults.

next.

**Comment by jessegreenberg** [comment]
How about TwoColumnHelpContent? Just an HBox with appropriate defaults. This is the layout for pretty much all of our KeyboardHelpDialogs.

**Comment by pixelzoom** [comment]
> How about TwoColumnHelpContent? Just an HBox with appropriate defaults.

If "just an HBox" means `extends HBox`), then you'd need to somehow limit the number of children to 2.   So this is a case where composition might be better, so that you can hide HBox's API. E.g.

```js
class TwoColumnHelpContent extends Node {

  constructor( leftNode, rightNode, options ) {

    options = _.extend( {
      ...
    }, options );

    const hBox = new HBox( {
      children: [ leftNode, rightNode ],
      ...
    } );
    
    assert && assert( !options.children, 'TwoColumnHelpContent sets children' );
    options = _.extend( {
      children: [ hBox ]
    }, options );
    super( options );
  }
  ...
}
```

**Comment by jessegreenberg** [comment]
I like it a lot, that is what I had in mind. Thanks!

**Comment by jessegreenberg** [comment]
OK, I think this is done. Sorry, to be cleaner I should have made another issue for the new type. @zepumph could you please review `TwoColumnKeyboardHelpContent`? Feel free to spot check changes for https://github.com/phetsims/scenery-phet/issues/478#issuecomment-479077697 if you wish but I don't feel it is as necessary.

**Comment by zepumph** [comment]
These changes are so great. They are really overdue, so thank you!

* I really like that TwoColumnKeyboardHelpContent factors out the duplicated spacing and so on for the HBox. What if we took this one step farther. It would be nice if `TwoColumnKeyboardHelpContent` could take parameters that were `KeyboardHelpSection` or lists of `KeyboardHelpSection` to be formed into VBoxes instead of Nodes. Perhaps then we could have even more of the alignment factored out into this file. I feel this way in general because I feel like Help dialog instances shouldn't likely need to adjust spacing themselves, there can probably be a consistent look/layout for all of them. 

Otherwise things seem nice! I took this for a test drive in GFLB the other day for tweaks there, and things are super pleasant. Thank you again.

**Comment by jessegreenberg** [comment]
Thanks for the review! That was a great suggestion, a number of VBoxes could be removed. @zepumph could you please review again, was this what you had in mind?

**Comment by zepumph** [comment]
That looks excellent. Thank you. Especially https://github.com/phetsims/gravity-force-lab/commit/f48f5a9fbc8d2001817eac47b08db742fabb5790, very prime!

Back to you.

**Comment by jessegreenberg** [comment]
Thanks for reviewing @zepumph, then https://github.com/phetsims/scenery-phet/issues/478#issuecomment-479077697 is done with the additional cleanup above and this can be closed.
